### PR TITLE
format: add checks for type aliases

### PR DESCRIPTION
**Pull request by tomocy**
Signed-off-by: tomocy <tomocy.dev@gmail.com>

Commit Message: format: add checks for type aliases
Additional Description: N/A
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

Part of  #11634

### CODE REVIEW

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -175,6 +175,21 @@
...
+    "^(.*::){,1}(StrictMock<|NiceMock<).*$",
+    "^(.*::){,1}(Test|Mock|Fake).*$",
+    "^Protobuf.*::.*$",
+    "^[A-Z]$",
+    "^.*, .*",
+    r"^.*\[\]$",
+}
+
+USING_TYPE_ALIAS_REGEX = re.compile("(using .* = .*;|typedef .* .*;)")
+SMART_PTR_REGEX = re.compile("std::(unique_ptr|shared_ptr)<(.*?)>(?!;)")
```
Comment (Shikugawa) [inline tools/code_format/check_format.py:189]: Should we introduce type alias for `weak_ptr`? This kind of smart pointer is partly used in envoyproxy/envoy and envoy/envoy-wasm.

Comment (jmarantz) [reply tools/code_format/check_format.py:189]: Sure, but I'd recommend not expanding the scope of this series of PRs. Would be an easier follow-up, as there are relatively fewer weak_ptr uses I think.

File: tools/code_format/check_format.py [hunk 2]
```diff
@@ -989,6 +1087,9 @@ def checkErrorMessages(error_messages):
                       type=str,
                       default=",".join(common.includeDirOrder()),
                       help="specify the header block include directory order.")
+  parser.add_argument("--aggressive",
+                      action='store_true',
+                      help="specify if it fixes formats making risky changes.")
```
Comment (jmarantz) [inline tools/code_format/check_format.py:1092]: help="makes risky format changes; re-testing needed after making aggressive fixes"

**Comment by jmarantz** [comment]
Please see @snowp's comment in slack: https://envoyproxy.slack.com/archives/C78HA81DH/p1594908964168900

I'm not sure that deep-link works properly but it's in #envoy-dev and @snowp wrote it on July 16th, and it starts with "just a thought i had: would it make sense to be using templated "

**Pull request closed without merging**